File: 305/solution.ts
function depthJSON(json: any): number {
	return (typeof json != 'object' && json != null) ? 0 : 1 + Math.max(0, ...Object.values(json).map((val) => depthJSON(val) as number));
}


console.log(depthJSON({ "a": 1 }))
// 1
console.log(depthJSON([1, 2, 3, 4, 5]))
// 1
console.log(depthJSON([{ "a": [] }, ["abc"]]))
// 3
console.log(depthJSON("hi"))
// 0 

File: 305/solution.rs

fn main(){
	println!("{}",depth_json("[{ \"a\": [] }, [\"abc\"]]".to_string()))
}

fn depth_json(json: String) -> i32 {
	let mut depth = 0;
	let mut max_depth = 0;
	for c in json.chars(){
		if c == '{' || c == '['{
			depth += 1;
			max_depth = if depth > max_depth { depth } else { max_depth };
		}
		else if c == '}' || c == '['{
			depth -= 1;
		}
	}
	max_depth
}